### PR TITLE
Remove purescript-generics, upgrade dependencies

### LINKED ISSUE #73: Won't build with latest `purescript-colors` 4.3.0
using `purs` current `master` branch (prints `--version` as `0.11.6 dev build`), attempting to `pulp build` build a project with `purescript-css` (got version 3.3.0 from Pursuit) in the deps (and insisting on `purescript-colors` 4.3.0) results in:

    * Building project in /home/rox/c/ps/test-pscorefn-src
    Compiling CSS.String
    Compiling CSS.Property
    Compiling CSS.Selector
    Compiling CSS.Color
    Error 1 of 2:
    
      at bower_components/purescript-css/src/CSS/Selector.purs line 25, column 8 - line 25, column 54
      
        Cannot derive a type class instance for
                                        
          Data.Generic.Generic Predicate
                                        
        since instances of this type class are not derivable.
        
        
      See https://github.com/purescript/documentation/blob/master/errors/CannotDerive.md for more information,
      or to contribute content related to this error.
      
    Error 2 of 2:
    
      at bower_components/purescript-css/src/CSS/Property.purs line 20, column 8 - line 20, column 52
      
        Cannot derive a type class instance for
                                       
          Data.Generic.Generic Prefixed
                                       
        since instances of this type class are not derivable.

### PR DESCRIPTION
**Pull request by vyorkin**
Should fix #73.
I had to manually implement `Show` type class in several cases

### CODE REVIEW

File: src/CSS/ListStyle/Image.purs [hunk 1]
```diff
@@ -20,10 +19,13 @@ data ListStyleImage
 
 derive instance eqListStyleImage :: Eq ListStyleImage
 derive instance ordListStyleImage :: Ord ListStyleImage
-derive instance genericListStyleImage :: Generic ListStyleImage
 
 instance showListStyleImage :: Show ListStyleImage where
-  show = gShow
+  show (ListStyleImage url) = "(ListStyleImage \"" <> url <> "\")"
```
Comment (vyorkin) [inline src/CSS/ListStyle/Image.purs]: Is it ok that I use quotes here?

Comment (kRITZCREEK) [reply src/CSS/ListStyle/Image.purs]: It's easier (and more correct) to do:
`"(ListStyleImage " <> show url <> ")"`

as that will use the proper escapes for everything inside the string.

**Comment by kRITZCREEK** [comment]
We're probably going to assimilate this PR into a 0.12 PR (which might need a few more tweaks and more version bumps). Thanks for starting the work :+1:

File: src/CSS/ListStyle/Image.purs (hunk 1 above)
Comment (vyorkin) [reply src/CSS/ListStyle/Image.purs]: Thanks, I'll squash the commits

**Comment by safareli** [comment]
Would be nice to add `package-lock.json ` to `.gitignore` too

**Comment by vyorkin** [comment]
@safareli sure, but I thought it is intended to be committed, [according to official docs](https://github.com/npm/npm/blob/v5.0.0/doc/files/package-lock.json.md)

**Comment by vyorkin** [comment]
@safareli added 📦-lock.json to gitignore

**Comment by vyorkin** [comment]
Rebased onto the latest `master` & squashed the commits. Is there anything that I can do in order to make it look better?

**Comment by kRITZCREEK** [comment]
This looks good, thanks!

**Pull request merged**